clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -120,6 +120,13 @@ public synchronized long timeToNextUpdate(long nowMs) {
     }
 
     /**
+     * The metadata has expired if an update is explicitly requested or an update is due now.
+     */
+    public synchronized  boolean hasExpired(long nowMs) {
+        return this.needUpdate || this.timeToNextUpdate(nowMs) == 0;
```

Comment:
It seems that the body of this method is equivalent to the following, which is already used in the method `timeToNextUpdate(..)`. This kind of duplicates the same log in two methods.

```
long timeToExpire = needUpdate ? 0 : Math.max(this.lastSuccessfulRefreshMs + this.metadataExpireMs - nowMs, 0);
return timeToExpire == 0;
```